### PR TITLE
Change overlay cache database interface

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: pkg/audit/cursor_test.go [hunk 1]
```diff
@@ -38,10 +38,17 @@ func TestAuditSegment(t *testing.T) {
...
+	defer tctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 4, 1)
+	require.NoError(t, err)
+	defer tctx.Check(planet.Shutdown)
+
+	planet.Start(tctx)
+
+	// we wait a second for all the nodes to complete bootstrapping off the satellite
+	time.Sleep(2 * time.Second)
```
Comment (thepaul) [inline pkg/audit/cursor_test.go]: Timed waits in tests always end up being fragile. Is there not any way to tell when all the nodes have completed bootstrapping?

File: satellite/satellitedb/overlaycache.go [hunk 2]
```diff
@@ -6,122 +6,234 @@ package satellitedb
...
-			updateFields,
-		)
+		infos[i] = info
+	}
+	return infos, nil
+}
+
+// List lists nodes starting from cursor
+func (cache *overlaycache) List(ctx context.Context, cursor storj.NodeID, limit int) ([]*pb.Node, error) {
+	dbxInfos, err := cache.db.Limited_OverlayCacheNode_By_NodeId_GreaterOrEqual(ctx,
```
Comment (mniewrzal) [inline satellite/satellitedb/overlaycache.go]: I think we should cover invalid `limit` with min and max.

File: pkg/audit/cursor_test.go (hunk 1 above)
Comment (egonelbre) [reply pkg/audit/cursor_test.go]: Unfortunately, not yet. Yeah, I don't like it either, but there will be PR-s eventually to fix this.

Comment (thepaul) [reply pkg/audit/cursor_test.go]: 👍

File: satellite/satellitedb/overlaycache.go (hunk 2 above)
Comment (egonelbre) [reply satellite/satellitedb/overlaycache.go]: done

Comment (egonelbre) [reply satellite/satellitedb/overlaycache.go]: also created a separate task for reviewing the limit situation in the future

**Pull request merged**